pkg/release/spec.go

```diff
@@ -0,0 +1,11 @@
+package release
+
+import "context"
+
+type ConditionFunc func() error
+
+type ConditionSet interface {
+	// SecretExists return a function waiting for the Secret to appear in the
+	// Kubernetes API.
+	SecretExists(ctx context.Context, namespace, name string) ConditionFunc
+}
```

Comment:
I don't like moving that to spec.go. This isn't package spec. But fine.

`PodExists` is missing here.

Reply:
I see you added it in the meantime. Please remove `CRDExists`. We don't use it outside and it wasn't exposed on purpose. With `CRDExists` this interface doesn't make sense as it hides nothing and isn't used for testing.

Reply:
I do not follow why we should hide here anything. All the conditions are used for the test setup. Why hiding any of them? I do not follow the vision here.

Reply:
The api is wierd here. It requires `crd *apiextensionsv1beta1.CustomResourceDefinition`. We can decide to change thing w/o all vendor updates we'd need otherwise. That's why I'm extra careful with exposing anything.

Reply:
I see that we have `InstallOperator` which takes a `CRD` and does the condition magic internally. I do not see the reason to hide the condition. The CRD is provided as argument anyway so we could just accept the condition. Then we can maybe even get rid of `InstallOperator` and just use `Install` with the proper condition. So you would configure it with the CRD and pod condition for the operator and do the logging magic separate. This would be clean and not hide anything but be composable. This would then be done in `e2esetup` later anyway.

Reply:
`InstallOperator` does also some logging magic, if not that I would agree here to get rid of `InstallOperator`. Maybe at some point we decide to do it like that. I'm not so keen to remove it right now when we have so many changes already.

The reason to hide `CRDExists` it to make sure it's not used outside `e2e-harness` repo. This gives flexibility of changing internal code w/o worrying about vendoring. It's easy to expose it when needed. It might be hard to hide it again. I think this is a very basic programming principle.

Reply:
Disagree and commit.

Reply:
Removed `CRDExists` from the interface.

Reply:
No reason to be sad. We can add it any time. I disagreed and committed in this PR also. We are equal :D